core/services/relay/evm/chain_reader.go

```diff
@@ -87,12 +106,21 @@ func (cr *chainReader) GetLatestValue(ctx context.Context, bc commontypes.BoundC
 }
 
 func (cr *chainReader) Start(ctx context.Context) error {
-	if err := cr.initialize(); err != nil {
-		return fmt.Errorf("Failed to initialize ChainReader: %w", err)
+	for name, eventId := range cr.events {
+		if err := cr.lp.RegisterFilter(logpoller.Filter{
```

Comment:
Can this be done in in the background so it doesn't block start up? (or cause it to fail completely)
For a service like this, we typically embed `services.StateMachine` and use the helpers to bootstrap things.